### PR TITLE
Avaya VSP Temperature Support

### PR DESCRIPTION
**Pull request by ospfbgp**
Avaya VSP Operating System Software Temperature Support

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by ospfbgp** [comment]
Code is complete and tested on multiple models of Avaya VSP.

File: includes/discovery/sensors/temperature/avaya-vsp.inc.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Copyright (c) 2017 Daniel Cox <danielcoxman@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+     
+$high_limit = 55;
```
Comment (laf) [inline includes/discovery/sensors/temperature/avaya-vsp.inc.php]: Why set hard limits? If the device doesn't provide them then we calculate the high / low limits.

File: includes/discovery/sensors/temperature/avaya-vsp.inc.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+$low_limit = 0;
+     
+// rcChasFanAmbientTemperature
+$tempsfan = snmp_walk($device, '.1.3.6.1.4.1.2272.1.4.7.1.1.3', '-Osqn');
```
Comment (laf) [inline includes/discovery/sensors/temperature/avaya-vsp.inc.php]: You should change your walks to use snmpwalk_cache_*, they provide the data in an array so you don't need to do explodes and things.

File: includes/discovery/sensors/temperature/avaya-vsp.inc.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        //rcSingleCpSystemMac2Temperature
+        case ($oid == '.1.3.6.1.4.1.2272.1.212.5.0' && $t[1] !=0):
+            $var1 = "rcSingleCpSystemMac2Temperature";
+            $var2 = "MAC2 temperature";
+            discover_sensor($valid['sensor'], 'temperature', $device, $oid, $var1.$i, $var1, $var2, '1', '1', $low_limit, $low_limit, $high_limit, $high_limit, $val);
+            break;
+        default:
+            $var1 = "Unknown";
+    }
+}
```
Comment (laf) [inline includes/discovery/sensors/temperature/avaya-vsp.inc.php]: Please unset() the two variables you've created from the walks so memory allocation is reduced slightly.

**Comment by ospfbgp** [comment]
I know that low limit just seems to get random numbers and so I just set it at 0 so that the graphs look uniform and you have a starting value to look at.  I have had some issues with switches that were too hot.  55 is the upper end of what they can handle.  I want to be alerted if anything hits 55 C.  It seems to be the magic high limit for the vsp switches.

I will fix the code and make it like you say.  It will take me a day or two and I want to test it in my lab.

**Comment by ospfbgp** [comment]
New code submitted.   Hope it is up to the task.  :-)

**Comment by laf** [comment]
I've made some small changes, will merge once checks complete.

**Pull request merged**